modules/cardano-faucet.nix

```diff
@@ -0,0 +1,55 @@
+with (import ./../lib.nix);
+
+globals: params:
+{ config, pkgs, lib, ...}:
+
+let
+  recaptcha = {
+    siteKey = lib.fileContents ../static/recaptcha_site_key;
+    secretKey = lib.fileContents ../static/recaptcha_secret_key;
+  };
+  explorerURL = "http://cardano-explorer.cardano-testnet.iohkdev.io/";
+  faucetFrontend = pkgs.runCommand "faucet-frontend" {} ''
+    mkdir -p $out
+    sed -e 's|recaptchaSiteKey:.*|recaptchaSiteKey: "${recaptcha.siteKey}",|g' \
+        -e 's|explorerURL:.*|explorerURL: "${explorerURL}",|g' \
+        ${../lib/faucet.html} > $out/index.html
+  '';
+
+in {
+  global.dnsHostname = mkForce "cardano-faucet";
+
+  imports = [ ./cardano-faucet-module.nix ];
+
+  services.cardano-faucet = {
+    enable = true;
+    relayHost = "relays.cardano-testnet.iohkdev.io";
+    configKey = "testnet_full";
+    statsd.enable = true;
+    statsd.port = 8125; # dogstatsd will be listening on this port
+    inherit recaptcha;
+  };
+
+  networking.firewall.allowedTCPPorts = [ 80 ];
+
+  services.nginx = {
+    enable = true;
+    virtualHosts = let
+      vhostDomainName = if config.global.dnsDomainname != null
+        then config.global.dnsDomainname else "iohkdev.io";
+     in {
+      "cardano-faucet.${vhostDomainName}" = {
+        # TLS provided by cloudfront
+        locations = {
+          "/" = {
+            root = faucetFrontend;
+            tryFiles = "$uri $uri/index.html @faucet";
```

Comment:
what about error pages? What do we want the user to see if the backend isn't available, or takes to long to respond, etc...

Reply:
The standard horrible error pages from NGINX should be OK.

This is because the faucet will normally be called by XHR through a page like this https://testnet.iohkdev.io/goguen/faucet/ . This frontend should show some "unavailable" message when it gets 5xx responses.